src/wrappers/digift/DigiftWrapper.sol

```diff
@@ -0,0 +1,1118 @@
+     */
+    function globalPendingRedeemRequest() external view returns (uint256) {
+        return _globalState.pendingRedeemRequest;
+    }
+
+    /**
+     * @notice Get the pending deposit request for a specific controller
+     * @param requestId The request ID (unused in this implementation)
+     * @param controller The controller address
+     * @return The pending deposit request amount
+     */
+    function pendingDepositRequest(uint256 requestId, address controller) external view returns (uint256) {
+        return _nodeState[controller].pendingDepositRequest;
+    }
+
+    /**
+     * @notice Get the claimable deposit request for a specific controller
+     * @param requestId The request ID (unused in this implementation)
+     * @param controller The controller address
+     * @return The claimable deposit request amount
+     */
+    function claimableDepositRequest(uint256 requestId, address controller) external view returns (uint256) {
+        return _nodeState[controller].claimableDepositRequest;
+    }
+
+    /**
+     * @notice Get the pending redemption request for a specific controller
+     * @param requestId The request ID (unused in this implementation)
+     * @param controller The controller address
+     * @return The pending redemption request amount
+     */
+    function pendingRedeemRequest(uint256 requestId, address controller) external view returns (uint256) {
+        return _nodeState[controller].pendingRedeemRequest;
+    }
+
+    /**
+     * @notice Get the claimable redemption request for a specific controller
+     * @param requestId The request ID (unused in this implementation)
+     * @param controller The controller address
+     * @return The claimable redemption request amount
+     */
+    function claimableRedeemRequest(uint256 requestId, address controller) external view returns (uint256) {
+        return _nodeState[controller].claimableRedeemRequest;
+    }
+
+    // =============================
+    //    Interface Implementations
+    // =============================
+
+    /**
+     * @notice Check if contract supports a specific interface
+     * @dev Implements IERC165, IERC7575, IERC7540Deposit, and IERC7540Redeem
+     * @param interfaceId The interface ID to check
+     * @return True if the interface is supported
+     */
+    function supportsInterface(bytes4 interfaceId) external pure override returns (bool) {
+        return interfaceId == type(IERC165).interfaceId || interfaceId == type(IERC7575).interfaceId
+            || interfaceId == type(IERC7540Deposit).interfaceId || interfaceId == type(IERC7540Redeem).interfaceId;
+    }
+
+    // =============================
+    //    ERC4626 Compatibility
+    // =============================
+
+    /**
+     * @notice Get the total assets under management
+     * @dev Calculates total assets based on current share price
+     * @return The total assets value
+     */
+    function totalAssets() external view returns (uint256) {
```

Comment:
It will revert if price data is outdated or has spiked significantly to prevent potential misaccounting on the Node contract.